synapse/metrics/__init__.py

```diff
@@ -53,7 +66,7 @@
 
 class RegistryProxy:
     @staticmethod
-    def collect():
+    def collect() -> Iterable[Metric]:
```

Comment:
I would think this needs to be a `Generator`, but maybe not?

This is true for a variety of spots we use `yield`.

Reply:
(Or maybe I can significantly simplify type hints in other spots!)

Reply:
I opted for using the most basic type that callers want, to minimize assumptions, rather than documenting the concrete return type.